how-to-build-gcc-from-source.rst

```diff
@@ -0,0 +1,133 @@
+How to Build GCC from Source
+============================
+
+Step 1: Install Dependancies
+----------------------------
+
+.. code-block:: console
+
+   sudo dnf install gcc-c++ \
+	libmpc-devel \
+	mpfr-devel \
+	gmp-devel \
+	bison \
+	flex \
+	dejagnu \
+	zlib-devel \
+	glibc-devel.i686
+
+Step 2: Set up Directories
+--------------------------
+
+Development for GCC is split into two different directories: A source, and 
+a build. The source directory is where all the source files will go. This is
+where you will clone the git repo, and where you will be changing files for 
+development. The build directory is where you are going to configure and
+build gcc. This directory contains all of the object files. After every
+edit to the source directory, you will have to recompile the object files.
+
+.. code-block:: console
+
+   mkdir -p ~/bld/trunk && mkdir ~/src/
+
+Step 2a: Clone Git Repo into Source Directory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Make a build and source directory. Here I will make it in my home directory;
+however, you can make the directories where you choose.
+
+.. code-block:: console
+
+   cd ~/src/ && git clone git://gcc.gnu.org/git/gcc.git
+
+Then you will want to checkout branch ``trunk`` or ``master``
+
+.. code-block:: console
+
+   git checkout master
+
+.. note::
+
+   If your git username and email are not the same as the one you use to
+   contribute to GCC (i.e. work email), it's helpful to set it to the correct
+   information inside the git directory.
+
+   .. code-block:: console
+
+      git config user.name "User Name"
+      git config user.email "email@address.com"
+
+   This is useful for when you are generating patches later, because the script
+   grabs your git information. This means you will not have to change it every
+   time you make a new patch.
+
+Step 2b:  Set up Build Directory Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+GCC has a script called ``configure`` inside the source directory at
+``~/gcc/configure``. Calling by itself will enable every option by default.
```

Comment:
Does it really "enable every option by default" ?   IIRC it picks defaults for every option, which is somewhat different.